File: src/engine/gameEngine.ts
// Re-export types, constants, and board operations for backward compatibility.
// Consumers can import from these sub-modules directly for more granular access.
export type { Piece, Turn, Phase, Position, GameMode, Difficulty, GameConfig, GameMove, GameState } from './types';
export { BOARD_SIZE, TOTAL_SHEEP, SHEEP_TO_WIN } from './constants';
export { isInBounds, hasDiagonals, getNeighbors, getCaptureTargets, getValidMovesForPiece } from './boardOps';

import { Piece, Turn, Phase, Position, GameMove, GameState } from './types';
import { BOARD_SIZE, TOTAL_SHEEP, SHEEP_TO_WIN } from './constants';
import { getNeighbors, getCaptureTargets, getValidMovesForPiece } from './boardOps';

function checkKittensBlocked(board: Piece[][]): boolean {
  for (let r = 0; r < BOARD_SIZE; r++) {
    for (let c = 0; c < BOARD_SIZE; c++) {
      if (board[r][c] === 'KITTY') {
        const moves = getValidMovesForPiece(board, r, c, 'KITTY');
        if (moves.length > 0) return false;
      }
    }
  }
  return true;
}

function checkSheepHaveMoves(board: Piece[][]): boolean {
  for (let r = 0; r < BOARD_SIZE; r++) {
    for (let c = 0; c < BOARD_SIZE; c++) {
      if (board[r][c] === 'SHEEP') {
        const neighbors = getNeighbors(r, c);
        for (const [nr, nc] of neighbors) {
          if (board[nr][nc] === 'EMPTY') return true;
        }
      }
    }
  }
  return false;
}

export function createInitialState(): GameState {
  const board: Piece[][] = Array.from({ length: BOARD_SIZE }, () =>
    Array(BOARD_SIZE).fill('EMPTY')
  );
  // Place 4 kittens at corners
  board[0][0] = 'KITTY';
  board[0][4] = 'KITTY';
  board[4][0] = 'KITTY';
  board[4][4] = 'KITTY';

  return {
    board,
    turn: 'SHEEP',
    phase: 'PLACEMENT',
    sheepPlaced: 0,
    sheepCaptured: 0,
    selectedPiece: null,
    winner: null,
    forfeitedBy: null,
    validMoves: [],
    lastMove: null,
  };
}

export function handleTap(state: GameState, row: number, col: number): GameState {
  if (state.winner) return state;

  const newBoard = state.board.map(r => [...r]);

  // SHEEP's turn
  if (state.turn === 'SHEEP') {
    if (state.phase === 'PLACEMENT') {
      // Place a sheep on empty intersection
      if (newBoard[row][col] !== 'EMPTY') return state;
      newBoard[row][col] = 'SHEEP';
      const newSheepPlaced = state.sheepPlaced + 1;
      const newPhase = newSheepPlaced >= TOTAL_SHEEP ? 'MOVEMENT' : 'PLACEMENT';
      const move: GameMove = { type: 'place', to: [row, col] };

      let winner: Turn | null = null;
      if (checkKittensBlocked(newBoard)) {
        winner = 'SHEEP';
      }

      return {
        board: newBoard,
        turn: 'KITTY',
        phase: newPhase,
        sheepPlaced: newSheepPlaced,
        sheepCaptured: state.sheepCaptured,
        selectedPiece: null,
        winner,
        forfeitedBy: null,
        validMoves: [],
        lastMove: move,
      };
    } else {
      // MOVEMENT phase - select then move
      if (state.selectedPiece === null) {
        // Select a sheep
        if (newBoard[row][col] !== 'SHEEP') return state;
        const moves = getValidMovesForPiece(newBoard, row, col, 'SHEEP');
        if (moves.length === 0) return state;
        return {
          ...state,
          selectedPiece: [row, col],
          validMoves: moves,
        };
      } else {
        const [sr, sc] = state.selectedPiece;
        // Tap same piece to deselect
        if (row === sr && col === sc) {
          return { ...state, selectedPiece: null, validMoves: [] };
        }
        // Tap another sheep to re-select
        if (newBoard[row][col] === 'SHEEP') {
          const moves = getValidMovesForPiece(newBoard, row, col, 'SHEEP');
          if (moves.length === 0) return state;
          return {
            ...state,
            selectedPiece: [row, col],
            validMoves: moves,
          };
        }
        // Check if this is a valid move
        const isValid = state.validMoves.some(([mr, mc]) => mr === row && mc === col);
        if (!isValid) return state;

        newBoard[row][col] = 'SHEEP';
        newBoard[sr][sc] = 'EMPTY';
        const move: GameMove = { type: 'move', from: [sr, sc], to: [row, col] };

        let winner: Turn | null = null;
        if (checkKittensBlocked(newBoard)) {
          winner = 'SHEEP';
        }

        return {
          board: newBoard,
          turn: 'KITTY',
          phase: state.phase,
          sheepPlaced: state.sheepPlaced,
          sheepCaptured: state.sheepCaptured,
          selectedPiece: null,
          winner,
          forfeitedBy: null,
          validMoves: [],
          lastMove: move,
        };
      }
    }
  }

  // KITTY's turn
  if (state.turn === 'KITTY') {
    if (state.selectedPiece === null) {
      // Select a kitty
      if (newBoard[row][col] !== 'KITTY') return state;
      const moves = getValidMovesForPiece(newBoard, row, col, 'KITTY');
      if (moves.length === 0) return state;
      return {
        ...state,
        selectedPiece: [row, col],
        validMoves: moves,
      };
    } else {
      const [sr, sc] = state.selectedPiece;
      // Tap same kitty to deselect
      if (row === sr && col === sc) {
        return { ...state, selectedPiece: null, validMoves: [] };
      }
      // Tap another kitty to re-select
      if (newBoard[row][col] === 'KITTY') {
        const moves = getValidMovesForPiece(newBoard, row, col, 'KITTY');
        if (moves.length === 0) return state;
        return {
          ...state,
          selectedPiece: [row, col],
          validMoves: moves,
        };
      }
      // Check if valid move
      const isValid = state.validMoves.some(([mr, mc]) => mr === row && mc === col);
      if (!isValid) return state;

      // Is it a capture?
      const captures = getCaptureTargets(state.board, sr, sc);
      const capture = captures.find(cap => cap.to[0] === row && cap.to[1] === col);

      newBoard[row][col] = 'KITTY';
      newBoard[sr][sc] = 'EMPTY';

      let newCaptured = state.sheepCaptured;
      let move: GameMove;

      if (capture) {
        newBoard[capture.captured[0]][capture.captured[1]] = 'EMPTY';
        newCaptured++;
        move = { type: 'capture', from: [sr, sc], to: [row, col], captured: capture.captured };
      } else {
        move = { type: 'move', from: [sr, sc], to: [row, col] };
      }

      let winner: Turn | null = null;
      if (newCaptured >= SHEEP_TO_WIN) {
        winner = 'KITTY';
      } else if (checkKittensBlocked(newBoard)) {
        winner = 'SHEEP';
      }

      // Check if sheep have any moves (in movement phase)
      const newPhase = state.sheepPlaced >= TOTAL_SHEEP ? 'MOVEMENT' : 'PLACEMENT';
      if (newPhase === 'MOVEMENT' && !winner && !checkSheepHaveMoves(newBoard)) {
        // Sheep can't move - kittens win (stalemate)
        winner = 'KITTY';
      }

      return {
        board: newBoard,
        turn: 'SHEEP',
        phase: newPhase,
        sheepPlaced: state.sheepPlaced,
        sheepCaptured: newCaptured,
        selectedPiece: null,
        winner,
        forfeitedBy: null,
        validMoves: [],
        lastMove: move,
      };
    }
  }

  return state;
}

export function applyMove(state: GameState, move: GameMove): GameState {
  const newBoard = state.board.map(r => [...r]);
  let newSheepPlaced = state.sheepPlaced;
  let newSheepCaptured = state.sheepCaptured;
  let newPhase = state.phase;
  let nextTurn: Turn;

  if (move.type === 'place') {
    newBoard[move.to[0]][move.to[1]] = 'SHEEP';
    newSheepPlaced++;
    newPhase = newSheepPlaced >= TOTAL_SHEEP ? 'MOVEMENT' : 'PLACEMENT';
    nextTurn = 'KITTY';
  } else if (move.type === 'move') {
    const [fr, fc] = move.from!;
    const piece = newBoard[fr][fc];
    newBoard[move.to[0]][move.to[1]] = piece;
    newBoard[fr][fc] = 'EMPTY';
    nextTurn = state.turn === 'SHEEP' ? 'KITTY' : 'SHEEP';
  } else {
    // capture
    const [fr, fc] = move.from!;
    newBoard[move.to[0]][move.to[1]] = 'KITTY';
    newBoard[fr][fc] = 'EMPTY';
    newBoard[move.captured![0]][move.captured![1]] = 'EMPTY';
    newSheepCaptured++;
    nextTurn = 'SHEEP';
  }

  let winner: Turn | null = null;
  if (newSheepCaptured >= SHEEP_TO_WIN) {
    winner = 'KITTY';
  } else if (checkKittensBlocked(newBoard)) {
    winner = 'SHEEP';
  } else if (newPhase === 'MOVEMENT' && nextTurn === 'SHEEP' && !checkSheepHaveMoves(newBoard)) {
    winner = 'KITTY';
  }

  return {
    board: newBoard,
    turn: nextTurn,
    phase: newPhase,
    sheepPlaced: newSheepPlaced,
    sheepCaptured: newSheepCaptured,
    selectedPiece: null,
    winner,
    forfeitedBy: null,
    validMoves: [],
    lastMove: move,
  };
}

export function forfeitGame(state: GameState): GameState {
  if (state.winner) return state;
  const forfeitingPlayer = state.turn;
  const opponent: Turn = forfeitingPlayer === 'SHEEP' ? 'KITTY' : 'SHEEP';
  return {
    ...state,
    winner: opponent,
    forfeitedBy: forfeitingPlayer,
    selectedPiece: null,
    validMoves: [],
  };
}

export function getGameStatusText(state: GameState): string {
  if (state.forfeitedBy === 'SHEEP') return 'Sheeps forfeited! Kittens win!';
  if (state.forfeitedBy === 'KITTY') return 'Kittens forfeited! Sheeps win!';
  if (state.winner === 'SHEEP') return 'Sheeps win! All kittens are blocked!';
  if (state.winner === 'KITTY') return 'Kittens win! Captured 5 sheeps!';

  if (state.turn === 'SHEEP') {
    if (state.phase === 'PLACEMENT') {
      return `Sheep's turn - Place a sheep (${state.sheepPlaced}/${TOTAL_SHEEP})`;
    }
    if (state.selectedPiece) return "Sheep's turn - Tap where to move";
    return "Sheep's turn - Select a sheep to move";
  }

  if (state.selectedPiece) return "Kitty's turn - Tap where to move";
  return "Kitty's turn - Select a kitty";
}

File: src/SheepsAndKittens.Core/Models/GameEngine.cs
using System.Collections.Generic;

namespace SheepsAndKittens.Core.Models
{
    public static class GameEngine
    {
        public const int BoardSize = 5;
        public const int TotalSheep = 20;
        public const int SheepToWin = 5;

        private static bool IsInBounds(int r, int c) =>
            r >= 0 && r < BoardSize && c >= 0 && c < BoardSize;

        public static bool HasDiagonals(int r, int c) =>
            (r + c) % 2 == 0;

        public static List<Position> GetNeighbors(int r, int c)
        {
            var neighbors = new List<Position>();

            // Orthogonal
            int[][] orthoDirs = { new[] { -1, 0 }, new[] { 1, 0 }, new[] { 0, -1 }, new[] { 0, 1 } };
            foreach (var dir in orthoDirs)
            {
                int nr = r + dir[0], nc = c + dir[1];
                if (IsInBounds(nr, nc))
                    neighbors.Add(new Position(nr, nc));
            }

            // Diagonals (only if (r+c) is even)
            if (HasDiagonals(r, c))
            {
                int[][] diagDirs = { new[] { -1, -1 }, new[] { -1, 1 }, new[] { 1, -1 }, new[] { 1, 1 } };
                foreach (var dir in diagDirs)
                {
                    int nr = r + dir[0], nc = c + dir[1];
                    if (IsInBounds(nr, nc))
                        neighbors.Add(new Position(nr, nc));
                }
            }

            return neighbors;
        }

        public static List<CaptureTarget> GetCaptureTargets(Piece[,] board, int r, int c)
        {
            var targets = new List<CaptureTarget>();
            if (board[r, c] != Piece.Kitty) return targets;

            var allDirs = new List<int[]>
            {
                new[] { -1, 0 }, new[] { 1, 0 }, new[] { 0, -1 }, new[] { 0, 1 }
            };

            if (HasDiagonals(r, c))
            {
                allDirs.Add(new[] { -1, -1 });
                allDirs.Add(new[] { -1, 1 });
                allDirs.Add(new[] { 1, -1 });
                allDirs.Add(new[] { 1, 1 });
            }

            foreach (var dir in allDirs)
            {
                int dr = dir[0], dc = dir[1];
                int midR = r + dr, midC = c + dc;
                int destR = r + 2 * dr, destC = c + 2 * dc;

                if (IsInBounds(midR, midC) && IsInBounds(destR, destC) &&
                    board[midR, midC] == Piece.Sheep && board[destR, destC] == Piece.Empty)
                {
                    bool isDiagJump = dr != 0 && dc != 0;
                    if (!isDiagJump || HasDiagonals(midR, midC))
                    {
                        targets.Add(new CaptureTarget(
                            new Position(destR, destC),
                            new Position(midR, midC)));
                    }
                }
            }

            return targets;
        }

        public static GameState CreateInitialState()
        {
            var state = new GameState
            {
                Turn = Turn.Sheep,
                Phase = Phase.Placement,
                SheepPlaced = 0,
                SheepCaptured = 0,
                SelectedPiece = null,
                Winner = null,
                ForfeitedBy = null,
                LastMove = null
            };

            // Place 4 kittens at corners
            state.Board[0, 0] = Piece.Kitty;
            state.Board[0, 4] = Piece.Kitty;
            state.Board[4, 0] = Piece.Kitty;
            state.Board[4, 4] = Piece.Kitty;

            return state;
        }

        public static List<Position> GetValidMovesForPiece(Piece[,] board, int r, int c, Piece piece)
        {
            var moves = new List<Position>();
            var neighbors = GetNeighbors(r, c);
            foreach (var n in neighbors)
            {
                if (board[n.Row, n.Col] == Piece.Empty)
                    moves.Add(n);
            }

            if (piece == Piece.Kitty)
            {
                var captures = GetCaptureTargets(board, r, c);
                foreach (var cap in captures)
                    moves.Add(cap.To);
            }

            return moves;
        }

        private static bool CheckKittensBlocked(Piece[,] board)
        {
            for (int r = 0; r < BoardSize; r++)
                for (int c = 0; c < BoardSize; c++)
                    if (board[r, c] == Piece.Kitty)
                    {
                        var moves = GetValidMovesForPiece(board, r, c, Piece.Kitty);
                        if (moves.Count > 0) return false;
                    }
            return true;
        }

        private static bool CheckSheepHaveMoves(Piece[,] board)
        {
            for (int r = 0; r < BoardSize; r++)
                for (int c = 0; c < BoardSize; c++)
                    if (board[r, c] == Piece.Sheep)
                    {
                        var neighbors = GetNeighbors(r, c);
                        foreach (var n in neighbors)
                            if (board[n.Row, n.Col] == Piece.Empty) return true;
                    }
            return false;
        }

        public static GameState HandleTap(GameState state, int row, int col)
        {
            if (state.Winner.HasValue) return state;

            var newState = state.Clone();

            // SHEEP's turn
            if (state.Turn == Turn.Sheep)
            {
                if (state.Phase == Phase.Placement)
                {
                    if (newState.Board[row, col] != Piece.Empty) return state;
                    newState.Board[row, col] = Piece.Sheep;
                    newState.SheepPlaced = state.SheepPlaced + 1;
                    newState.Phase = newState.SheepPlaced >= TotalSheep ? Phase.Movement : Phase.Placement;
                    newState.Turn = Turn.Kitty;
                    newState.SelectedPiece = null;
                    newState.ValidMoves = new List<Position>();
                    newState.ForfeitedBy = null;
                    newState.LastMove = new GameMove(MoveType.Place, new Position(row, col));

                    if (CheckKittensBlocked(newState.Board))
                        newState.Winner = Turn.Sheep;

                    return newState;
                }
                else
                {
                    // MOVEMENT phase - select then move
                    if (!state.SelectedPiece.HasValue)
                    {
                        if (newState.Board[row, col] != Piece.Sheep) return state;
                        var moves = GetValidMovesForPiece(newState.Board, row, col, Piece.Sheep);
                        if (moves.Count == 0) return state;
                        var sel = state.Clone();
                        sel.SelectedPiece = new Position(row, col);
                        sel.ValidMoves = moves;
                        return sel;
                    }
                    else
                    {
                        var sp = state.SelectedPiece.Value;
                        // Tap same piece to deselect
                        if (row == sp.Row && col == sp.Col)
                        {
                            var desel = state.Clone();
                            desel.SelectedPiece = null;
                            desel.ValidMoves = new List<Position>();
                            return desel;
                        }
                        // Tap another sheep to re-select
                        if (newState.Board[row, col] == Piece.Sheep)
                        {
                            var moves = GetValidMovesForPiece(newState.Board, row, col, Piece.Sheep);
                            if (moves.Count == 0) return state;
                            var resel = state.Clone();
                            resel.SelectedPiece = new Position(row, col);
                            resel.ValidMoves = moves;
                            return resel;
                        }
                        // Check if valid move
                        bool isValid = false;
                        foreach (var m in state.ValidMoves)
                            if (m.Row == row && m.Col == col) { isValid = true; break; }
                        if (!isValid) return state;

                        newState.Board[row, col] = Piece.Sheep;
                        newState.Board[sp.Row, sp.Col] = Piece.Empty;
                        newState.Turn = Turn.Kitty;
                        newState.SelectedPiece = null;
                        newState.ValidMoves = new List<Position>();
                        newState.ForfeitedBy = null;
                        newState.LastMove = new GameMove(MoveType.Move, new Position(row, col), new Position(sp.Row, sp.Col));

                        if (CheckKittensBlocked(newState.Board))
                            newState.Winner = Turn.Sheep;

                        return newState;
                    }
                }
            }

            // KITTY's turn
            if (state.Turn == Turn.Kitty)
            {
                if (!state.SelectedPiece.HasValue)
                {
                    if (newState.Board[row, col] != Piece.Kitty) return state;
                    var moves = GetValidMovesForPiece(newState.Board, row, col, Piece.Kitty);
                    if (moves.Count == 0) return state;
                    var sel = state.Clone();
                    sel.SelectedPiece = new Position(row, col);
                    sel.ValidMoves = moves;
                    return sel;
                }
                else
                {
                    var sp = state.SelectedPiece.Value;
                    // Tap same kitty to deselect
                    if (row == sp.Row && col == sp.Col)
                    {
                        var desel = state.Clone();
                        desel.SelectedPiece = null;
                        desel.ValidMoves = new List<Position>();
                        return desel;
                    }
                    // Tap another kitty to re-select
                    if (newState.Board[row, col] == Piece.Kitty)
                    {
                        var moves = GetValidMovesForPiece(newState.Board, row, col, Piece.Kitty);
                        if (moves.Count == 0) return state;
                        var resel = state.Clone();
                        resel.SelectedPiece = new Position(row, col);
                        resel.ValidMoves = moves;
                        return resel;
                    }
                    // Check if valid move
                    bool isValid = false;
                    foreach (var m in state.ValidMoves)
                        if (m.Row == row && m.Col == col) { isValid = true; break; }
                    if (!isValid) return state;

                    // Is it a capture?
                    var captures = GetCaptureTargets(state.Board, sp.Row, sp.Col);
                    CaptureTarget? capture = null;
                    foreach (var cap in captures)
                        if (cap.To.Row == row && cap.To.Col == col) { capture = cap; break; }

                    newState.Board[row, col] = Piece.Kitty;
                    newState.Board[sp.Row, sp.Col] = Piece.Empty;

                    int newCaptured = state.SheepCaptured;
                    GameMove move;

                    if (capture != null)
                    {
                        newState.Board[capture.Captured.Row, capture.Captured.Col] = Piece.Empty;
                        newCaptured++;
                        move = new GameMove(MoveType.Capture, new Position(row, col),
                            new Position(sp.Row, sp.Col), capture.Captured);
                    }
                    else
                    {
                        move = new GameMove(MoveType.Move, new Position(row, col),
                            new Position(sp.Row, sp.Col));
                    }

                    newState.SheepCaptured = newCaptured;
                    newState.Turn = Turn.Sheep;
                    newState.Phase = state.SheepPlaced >= TotalSheep ? Phase.Movement : Phase.Placement;
                    newState.SelectedPiece = null;
                    newState.ValidMoves = new List<Position>();
                    newState.ForfeitedBy = null;
                    newState.LastMove = move;

                    if (newCaptured >= SheepToWin)
                        newState.Winner = Turn.Kitty;
                    else if (CheckKittensBlocked(newState.Board))
                        newState.Winner = Turn.Sheep;
                    else if (newState.Phase == Phase.Movement && !CheckSheepHaveMoves(newState.Board))
                        newState.Winner = Turn.Kitty;

                    return newState;
                }
            }

            return state;
        }

        public static GameState ApplyMove(GameState state, GameMove move)
        {
            var newState = state.Clone();
            int newSheepPlaced = state.SheepPlaced;
            int newSheepCaptured = state.SheepCaptured;
            Phase newPhase = state.Phase;
            Turn nextTurn;

            if (move.Type == MoveType.Place)
            {
                newState.Board[move.To.Row, move.To.Col] = Piece.Sheep;
                newSheepPlaced++;
                newPhase = newSheepPlaced >= TotalSheep ? Phase.Movement : Phase.Placement;
                nextTurn = Turn.Kitty;
            }
            else if (move.Type == MoveType.Move)
            {
                var from = move.From!.Value;
                var piece = newState.Board[from.Row, from.Col];
                newState.Board[move.To.Row, move.To.Col] = piece;
                newState.Board[from.Row, from.Col] = Piece.Empty;
                nextTurn = state.Turn == Turn.Sheep ? Turn.Kitty : Turn.Sheep;
            }
            else // Capture
            {
                var from = move.From!.Value;
                newState.Board[move.To.Row, move.To.Col] = Piece.Kitty;
                newState.Board[from.Row, from.Col] = Piece.Empty;
                newState.Board[move.Captured!.Value.Row, move.Captured!.Value.Col] = Piece.Empty;
                newSheepCaptured++;
                nextTurn = Turn.Sheep;
            }

            newState.Turn = nextTurn;
            newState.Phase = newPhase;
            newState.SheepPlaced = newSheepPlaced;
            newState.SheepCaptured = newSheepCaptured;
            newState.SelectedPiece = null;
            newState.ValidMoves = new List<Position>();
            newState.ForfeitedBy = null;
            newState.LastMove = move;
            newState.Winner = null;

            if (newSheepCaptured >= SheepToWin)
                newState.Winner = Turn.Kitty;
            else if (CheckKittensBlocked(newState.Board))
                newState.Winner = Turn.Sheep;
            else if (newPhase == Phase.Movement && nextTurn == Turn.Sheep && !CheckSheepHaveMoves(newState.Board))
                newState.Winner = Turn.Kitty;

            return newState;
        }

        public static GameState ForfeitGame(GameState state)
        {
            if (state.Winner.HasValue) return state;
            var newState = state.Clone();
            var opponent = state.Turn == Turn.Sheep ? Turn.Kitty : Turn.Sheep;
            newState.Winner = opponent;
            newState.ForfeitedBy = state.Turn;
            newState.SelectedPiece = null;
            newState.ValidMoves = new List<Position>();
            return newState;
        }

        public static string GetGameStatusText(GameState state)
        {
            if (state.ForfeitedBy == Turn.Sheep) return "Sheeps forfeited! Kittens win!";
            if (state.ForfeitedBy == Turn.Kitty) return "Kittens forfeited! Sheeps win!";
            if (state.Winner == Turn.Sheep) return "Sheeps win! All kittens are blocked!";
            if (state.Winner == Turn.Kitty) return "Kittens win! Captured 5 sheeps!";

            if (state.Turn == Turn.Sheep)
            {
                if (state.Phase == Phase.Placement)
                    return $"Sheep's turn - Place a sheep ({state.SheepPlaced}/{TotalSheep})";
                if (state.SelectedPiece.HasValue) return "Sheep's turn - Tap where to move";
                return "Sheep's turn - Select a sheep to move";
            }

            if (state.SelectedPiece.HasValue) return "Kitty's turn - Tap where to move";
            return "Kitty's turn - Select a kitty";
        }
    }

    public class CaptureTarget
    {
        public Position To { get; }
        public Position Captured { get; }

        public CaptureTarget(Position to, Position captured)
        {
            To = to;
            Captured = captured;
        }
    }
}
